fastqc: add 0.12.1, update patch files

A-N-Other opened the pull request:
Adding the latest version of `fastqc` (skipping `@0.12.0` in favour of the bugfix `@0.12.1` that immediately followed).

As of `@0.12:`, `fastqc` now makes use of `htsjdk.jar` to read input files, which required a new patch file. Patch files now named appropriately and are picked for use according to the version of `fastqc`.

Installed & tested with `openjdk@17.0.5_8`.

Commit:
fastqc: add 0.12.1

Commit:
fastqc: add 0.12.1

alecbcs commented on var/spack/repos/builtin/packages/fastqc/package.py:
```diff
@@ -29,8 +31,12 @@ def install(self, spec, prefix):
         install("fastqc", prefix.bin)
-        for j in ["cisd-jhdf5.jar", "jbzip2-0.9.jar", "sam-1.103.jar"]:
-            install(j, prefix.lib)
+        if self.spec.satisfies("@:0.11.9"):
+            for j in ["cisd-jhdf5.jar", "jbzip2-0.9.jar", "sam-1.103.jar"]:
+                install(j, prefix.lib)
+        else:
+            for j in ["cisd-jhdf5.jar", "jbzip2-0.9.jar", "htsjdk.jar"]:
+                install(j, prefix.lib)
```
```suggestion
        install("cisd-jhdf5.jar", prefix.lib)
        install("jbzip2-0.9.jar", prefix.lib)
        
        if self.spec.satisfies("@:0.11.9"):
                install("sam-1.103.jar", prefix.lib)
        else:
                install("htsjdk.jar", prefix.lib)
```

This isn't a hard requirement, but I typically opt for making packages easier to read when possible instead of going for maximum flexibility in the future. Otherwise this PR looks great!

A-N-Other committed:
Update var/spack/repos/builtin/packages/fastqc/package.py

Yeah, had considered doing the same, I'd just opted to maintain the status quo. All good.

Co-authored-by: Alec Scott <alec@bcs.sh>

A-N-Other committed:
Update package.py

Style fiddles to make the bot contented.

alecbcs reviewed (approved):
SHA confirmed. Looks good to me!

A-N-Other committed:
Merge branch 'develop' into fastqc
